tests/std/tests/P0896R4_ranges_alg_reverse_copy/test.cpp

```diff
@@ -0,0 +1,159 @@
+    ranges::reverse_copy_result<int*, int*>>);
+
+struct nontrivial_int {
+    int val;
+
+    constexpr nontrivial_int(int i = 0) noexcept : val{i} {}
+    constexpr nontrivial_int(const nontrivial_int& that) noexcept : val{that.val} {}
+    constexpr nontrivial_int& operator=(const nontrivial_int& that) noexcept {
+        val = that.val;
+        return *this;
+    }
+
+    auto operator<=>(const nontrivial_int&) const = default;
+};
+
+struct instantiator {
+    static constexpr nontrivial_int input[]    = {13, 42, 1367};
+    static constexpr nontrivial_int expected[] = {1367, 42, 13};
+
+    template <ranges::bidirectional_range In, indirectly_writable<ranges::range_reference_t<In>> Out>
+    static constexpr void call() {
+        if constexpr (!ranges::contiguous_range<In>) { // the vectorized tests below have plenty of contiguous coverage
+            using ranges::reverse_copy, ranges::reverse_copy_result, ranges::begin, ranges::end, ranges::equal,
+                ranges::iterator_t;
+
+            { // Validate iterator overload
+                nontrivial_int output[3];
+                In wrapped_input{input};
+                const same_as<reverse_copy_result<iterator_t<In>, Out>> auto result =
+                    reverse_copy(wrapped_input.begin(), wrapped_input.end(), Out{output});
+                assert(result.in == wrapped_input.end());
+                assert(result.out.peek() == end(output));
+                assert(equal(output, expected));
+            }
+            { // Validate range overload
+                nontrivial_int output[3];
+                In wrapped_input{input};
+                const same_as<reverse_copy_result<iterator_t<In>, Out>> auto result =
+                    reverse_copy(wrapped_input, Out{output});
+                assert(result.in == wrapped_input.end());
+                assert(result.out.peek() == end(output));
+                assert(equal(output, expected));
+            }
+
+            { // Validate iterator overload, empty range
+                nontrivial_int output[3];
+                In wrapped_input{};
+                const same_as<reverse_copy_result<iterator_t<In>, Out>> auto result =
+                    reverse_copy(wrapped_input.begin(), wrapped_input.end(), Out{output});
+                assert(result.in == wrapped_input.end());
+                assert(result.out.peek() == begin(output));
+            }
+            { // Validate range overload, empty range
+                nontrivial_int output[3];
+                In wrapped_input{};
+                const same_as<reverse_copy_result<iterator_t<In>, Out>> auto result =
+                    reverse_copy(wrapped_input, Out{output});
+                assert(result.in == wrapped_input.end());
+                assert(result.out.peek() == begin(output));
+            }
+        }
+    }
+};
+
+template <size_t N>
+struct bytes {
+    unsigned char storage[N];
+
+    constexpr bytes(unsigned char base = 0) {

```

Comment:
As far as I can see the only case where use the default value is for the output arrays. 

Could we use some sentinel values for that case instead of the iota such as -1?

Reply:
Sure. I don't much care what particular values we use as long as all bytes in the output initially have a value that doesn't appear in any byte of the input / expected result.